R/zchunk_L201.en_nonco2.R

```diff
@@ -0,0 +1,358 @@
+    A51.steepness %>%
+      gather(Non.CO2, value, SO2, NOx, CO, BC, OC, NMVOC) %>%
+      filter(supplysector == "out_resources") %>%
+      # extend steepness factors across all regions
+      repeat_add_columns(tibble(region = GCAM_region_names$region)) %>%
+      rename(depresource = subsector) ->
+      L201.steepness_res
+
+    L201.steepness_res %>%
+      select(region, depresource, Non.CO2) %>%
+      mutate(ctrl.name = "GDP_control") %>%
+      left_join(L201.steepness_res, by = c("region", "depresource", "Non.CO2")) %>%
+      na.omit %>%
+      select(region, depresource, Non.CO2, ctrl.name, steepness = value) ->
+      L201.nonghg_steepness_res
+
+    # Remove rows where we only have a value for one of max.reduction or steepness
+    # TODO: is this what we want or should we raise an error?
+    L201.nonghg_max_reduction_res %>%
+      full_join(L201.nonghg_steepness_res, by = c("region", "depresource", "Non.CO2", "ctrl.name")) %>%
+      na.omit %>%
+      # No need to include a GDP control when the max.reduction is zero
+      filter(max_reduction > 0) ->
+      L201.nonghg_gdp_control_res
+
+    L201.nonghg_max_reduction_res <- select(L201.nonghg_gdp_control_res, -steepness)
+    L201.nonghg_steepness_res <- select(L201.nonghg_gdp_control_res, -max_reduction)
+
+    # Rename to regional SO2
+    L201.en_pol_emissions <- rename_SO2(L201.en_pol_emissions, A_regions, FALSE)
+    L201.nonghg_max_reduction <- rename_SO2(L201.nonghg_max_reduction, A_regions, FALSE)
+    L201.nonghg_steepness <- rename_SO2(L201.nonghg_steepness, A_regions, FALSE)
+    L201.nonghg_res <- rename_SO2(L201.nonghg_res, A_regions, FALSE)
+    L201.nonghg_steepness_res <- rename_SO2(L201.nonghg_steepness_res, A_regions, FALSE)
+    L201.nonghg_max_reduction_res <- rename_SO2(L201.nonghg_max_reduction_res, A_regions, FALSE)
+
+    # Remove district heat from regions that do have have it
+    A_regions.en %>%
+      filter(has_district_heat == 1) %>%
+      .[["region"]] ->
+      distheat.regions
+    L201.en_pol_emissions <- filter(L201.en_pol_emissions, supplysector != "district heat" | region %in% distheat.regions)
+    L201.en_ghg_emissions <- filter(L201.en_ghg_emissions, supplysector != "district heat" | region %in% distheat.regions)
+    L201.en_bcoc_emissions <- filter(L201.en_bcoc_emissions, supplysector != "district heat" | region %in% distheat.regions)
+    L201.nonghg_max_reduction <- filter(L201.nonghg_max_reduction, supplysector != "district heat" | region %in% distheat.regions)
+    L201.nonghg_steepness <- filter(L201.nonghg_steepness, supplysector != "district heat" | region %in% distheat.regions)
+
+    # It may be the case with certain regional aggregations that regions exist that have no
+    # heating or cooling sectors. We should delete those here.
+    # Delete sectors that do not exist due to zero heating/cooling degree days
+    delete_zerodays <- function(x, L201.delete.sectors) {
```

Comment:
maybe a more informative function name? I think all regions have days in the year, just maybe not ones that need heating or cooling.